### ISSUE TITLE
VariableName check too strict around numbers

### ISSUE THREAD

**Issue by nikitaourazbaev** [Shopify/theme-tools#397]
**Describe the bug**
`first_3d_model` results in a `VariableName` check warning

**Source**

```liquid
{% assign first_3d_model = product.media | where: 'media_type', 'model' | first %}
```

**Expected behaviour**
`VariableName` check should pass

**Actual behaviour**
`VariableName` check doesn’t pass, with the warning message “The variable 'first_3d_model' uses wrong naming format”

**Debugging information**
 - macOS
 - 14.5
 - Shopify Liquid VS Code Extension v2.4.1

**Additional context**
I understand that `first_3_d_model` would be the correct casing according to the library used for the check, but in my opinion this goes against a long history of naming variables in Liquid, including in Shopify’s own themes. Adjusting the check so it’s more lenient about numbers would be preferred, especially as this check is on by default.

**Comment by madsenmm** [comment]
@nikitaourazbaev Added a fix in this PR:
https://github.com/Shopify/theme-tools/pull/462

**Comment by nikitaourazbaev** [comment]
> @nikitaourazbaev now merged: [#462 (review)](https://github.com/Shopify/theme-tools/pull/462#pullrequestreview-2208719136) Not sure how long before the update is available in the CLI.

@madsenmm Amazing, thank you so much!
